test/kafka-sink-stalled-errors/stalled.td

```diff
@@ -0,0 +1,42 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ kafka-create-topic topic=status-history
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+> CREATE CONNECTION IF NOT EXISTS csr_conn TO CONFLUENT SCHEMA REGISTRY (
+    URL '${testdrive.schema-registry-url}'
+  );
+
+> CREATE TABLE sink_table (f1 INTEGER);
+
+> CREATE SINK kafka_sink FROM sink_table
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-kafka-sink-${testdrive.seed}')
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+  ENVELOPE DEBEZIUM
+
+$ set-from-sql var=sink_id
+SELECT id FROM mz_sinks WHERE name = 'kafka_sink'
+
+$ set-regex match="\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d(\.\d\d\d)?" replacement="<TIMESTAMP>"
+
+# Verify we get a starting status
+> SELECT * FROM mz_internal.mz_sink_status_history WHERE sink_id = '${sink_id}' ORDER BY occurred_at LIMIT 1;
+"<TIMESTAMP> UTC" ${sink_id} starting <null> <null>
+
+> INSERT INTO sink_table VALUES (1);
+
+> INSERT INTO sink_table VALUES (2);
+
+# We should get a stalled error message next
+> SELECT status, sink_id, error, jsonb_object_keys(details) FROM mz_internal.mz_sink_status_history
```

Comment:
I don't want to randomize too much, but would it be possible to match on the actual hint string? Or maybe a "contains" filter? Might be harder than it's worth, in that. case, please ignore.

Reply:
I was actually trying to do a regex search, but [kept getting thrown off](https://github.com/MaterializeInc/materialize/pull/18009/commits/987a0df17e2c4a9177ad9466352c4fa813f79c74) by the double quotes in json. It just would not match properly and the `set-regex` I believe can't take raw strings (I could be missing something here)

Should be able to match on the actual string though. I did not think of it, because it's long 😅. Let me try that instead.

Reply:
@moulimukherjee how were you indexing into the jsonb object? I would select the `details->>'hint'` and just match that, instead of the full details column!

Reply:
Per Gus's feedback below, I will also restructure the string a bit to remove the linebreaks, will make it easier to exact match as well.

Reply:
@guswynn Ah, I was trying to match the entire `details` column which had the json string. Your suggestion is definitely much more saner. I am updating it to extract the json value instead.

Reply:
Updated!